src/Models/AZ3166Device.ts

```diff
@@ -95,10 +98,71 @@ export class AZ3166Device extends ArduinoDeviceBase {
     return true;
   }
 
+  async upload(): Promise<boolean> {
+    try {
+      if (!this.outputPath) {
+        vscode.window.showWarningMessage(
+            'No device compile output folder found.');
+        this.channel.show();
+        this.channel.appendLine('No device compile output folder found.');
+        return false;
+      }
+
+      const binFiles = fs.readdirSync(this.outputPath).filter(
+        file => path.extname(file).endsWith(constants.binFileExt) && !path.basename(file).endsWith(constants.otaBinFileExt));
+      // const binFiles = fs.listSync(this.outputPath, ['bin']);
+      if (!binFiles || !binFiles.length) {
+        const message =
+            'No bin file found. Please run the command of Device Compile first.';
+        vscode.window.showWarningMessage(message);
+        this.channel.show();
+        this.channel.appendLine(message);
+        return false;
+      }
+
+      try {
+        const hostVolumes = 
+        await sdk.FileSystem.listVolume();
+        const AZ3166Disk = hostVolumes.find(volume => {
+          const nameProperty = Object.getOwnPropertyDescriptor(volume, 'name');
+          if (nameProperty !== undefined && nameProperty.value === 'AZ3166') {
+            return true;
+          }
+          return false;
+        })
+
+        if (!AZ3166Disk) {
+          const message =
+          'No AZ3166 device found. Please plug in a devkit board.';
+          vscode.window.showWarningMessage(message);
+          this.channel.show();
+          this.channel.appendLine(message);
+          return false;
+        }
+
+        const binFilePath = path.join(this.outputPath, binFiles[0]);
+        const binFileHardCode = "C:\\Users\\dilin\\workspace\\deviceDevEx\\vscode-iot-workbench-github\\data\\devkit_project\\.build\\output.bin";

```

Comment:
The user profile path should not be hard code

Reply:
Test code. Forgot to remove it. Fixed.